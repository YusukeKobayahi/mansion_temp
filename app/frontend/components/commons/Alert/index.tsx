import styles from "./index.module.scss";
import classnames from "classnames";

type AlertProps = {
  text: string;
  children?: React.ReactNode;
  className?: string;
  color?: "brown" | "red";
  outline?: boolean;
};
const Alert: React.FC<AlertProps> = ({
  text,
  children,
  className,
  color = "red",
  outline = false,
}: AlertProps) => (
  <div
    className={classnames(
      styles.container,
      styles[color],
      { [styles.outline]: outline },
      className
    )}
  >
    <p>{text}</p>
    {children && <div className={styles.children}>{children}</div>}
  </div>
);

export default Alert;
