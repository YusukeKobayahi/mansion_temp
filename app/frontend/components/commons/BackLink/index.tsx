import Link from "next/link";
import { buildLibraryPagePath } from "~/lib/utils";
import styles from "./index.module.scss";

type BackLinkProps = {
  name: string;
  page: string;
  id: string;
};

const BackLink: React.FC<BackLinkProps> = ({
  name,
  page,
  id,
}: BackLinkProps) => {
  return (
    <Link href={buildLibraryPagePath(page, id)} passHref>
      <button className={styles.btn}>{name}の中古マンション一覧</button>
    </Link>
  );
};

export default BackLink;
