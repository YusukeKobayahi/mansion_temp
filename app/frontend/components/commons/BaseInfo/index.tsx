import styles from "./index.module.scss";
import UnderlineHeading from "~/components/commons/headings/UnderlineHeading";
import classnames from "classnames";
export interface BaseInfoProps {
  mainText: string;
  areaText: string;
  isNoindex: boolean;
  totalCount: number;
}

const BaseInfo: React.FC<BaseInfoProps> = ({
  mainText,
  areaText,
  isNoindex,
  totalCount,
}: BaseInfoProps) => {
  const explanation =
    totalCount == 0
      ? "条件にあう物件がありません。条件を変更して再度検索してください。"
      : `Housiiでは、「${
          isNoindex ? mainText + "のあなたの条件にあった」" : mainText + "」の"
        }中古マンションについて紹介しています。`;
  return (
    <div>
      <h1 className={styles.title}>
        <UnderlineHeading>
          <p className={"pcOnly"}>
            {isNoindex ? mainText + "のあなたの条件にあった" : mainText + "の"}
            中古マンション一覧
          </p>
          <p className={"spOnly"}>{areaText}の中古マンション一覧</p>
        </UnderlineHeading>
      </h1>
      <p className={classnames(styles.info, "pcOnly")}>{explanation}</p>
    </div>
  );
};

export default BaseInfo;
