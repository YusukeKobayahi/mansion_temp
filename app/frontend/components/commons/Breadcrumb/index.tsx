import Link from "next/link";
import styles from "~/components/commons/Breadcrumb/index.module.scss";

export type BreadcrumbItemProps = {
  name: string;
  path: string;
  asLink?: string;
};

type BreadcrumbProps = {
  breadcrumbs: BreadcrumbItemProps[];
  mainClassName?: string;
};

const Breadcrumb: React.FC<BreadcrumbProps> = ({
  breadcrumbs,
  mainClassName = "",
}: BreadcrumbProps) => {
  const items = breadcrumbs.map((b, i) => {
    if (b.name !== "" && b.path !== "") {
      return (
        <li key={i} className={styles.item}>
          {breadcrumbs.length - 1 === i ? (
            <span>{b.name}</span>
          ) : (
            <Link
              href={b.path}
              as={b.asLink ? b.asLink : b.path}
              prefetch={false}
            >
              <a>{b.name}</a>
            </Link>
          )}
        </li>
      );
    }
  });

  return (
    <nav className={mainClassName}>
      <ol className={styles.list}>
        <li className={styles.item}>
          <a href="/">イエウールトップ</a>
        </li>
        {items}
      </ol>
    </nav>
  );
};

export default Breadcrumb;
