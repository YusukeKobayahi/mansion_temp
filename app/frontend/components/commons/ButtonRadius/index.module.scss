@import "~/styles/variables";
@import "~/styles/utils";

.btn {
  width: 100%;
  display: block;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
  border-radius: 60px;
  cursor: pointer;

  &:focus {
    outline: none;
    box-shadow: none;
    opacity: 1;
  }
}

/* 
 * Color
 */

$dark: #746744;
$orange: #f15e3a;
$brown: $color-main;

@mixin fill($color: $brown) {
  background-color: $color;
  border: 2px solid $color;
  color: white;
}

@mixin outline($color: $brown) {
  &.outline {
    background-color: white;
    border: 2px solid $color;
    color: $color;
  }
}

@mixin disabled {
  &.disabled {
    background-color: #f4f4f4;
    border: 2px solid #e5e5e5;
    color: #e5e5e5;
    cursor: auto;
  }
}

.dark {
  @include fill($dark);
  @include outline($dark);
  @include disabled();
}

.orange {
  @include fill($orange);
  @include outline($orange);
  @include disabled();
}

.brown {
  @include fill;
  @include outline;
  @include disabled;
}

.md {
  line-height: 50px;
  font-size: 16px;
}
.sm {
  line-height: 20px;
  font-size: 12px;
}
