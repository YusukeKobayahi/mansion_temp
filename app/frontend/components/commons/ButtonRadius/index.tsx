import classnames from "classnames";
import styles from "./index.module.scss";
import NextLink, { LinkProps } from "next/link";
import ConditionalWrapper from "~/components/commons/ConditionalWrapper";

export type ButtonRadiusProps = {
  text: string | React.ReactElement;
  onClick?: React.MouseEventHandler;
  size?: "md" | "sm";
  color?: "dark" | "orange" | "brown";
  outline?: boolean;
  disabled?: boolean;
  className?: string;
  outerClassName?: string;
  linkProps?: LinkProps;
};

/**
 * ### linkProps
 * NextLinkコンポーネントに渡される引数の塊。指定しない場合はNextLinkは使用されない。
 * ### className
 * コンポーネント内のbuttonタグに直接当てられるクラス。
 * ### outerClassName
 * コンポーネント内の一番外側に当てられるクラス。
 * 基本的にmarginなどの外側に影響するスタイルを当てる際に使うといい。
 * NextLinkが使用されない場合はclassNameで完結できる場合がほとんどだが副作用に注意してスタイルを当てること。
 */
const ButtonRadius: React.FC<ButtonRadiusProps> = ({
  text,
  onClick,
  size = "md",
  color = "orange",
  outline = false,
  disabled = false,
  className = "",
  outerClassName = "",
  linkProps,
}: ButtonRadiusProps) => {
  const customStyles = classnames(
    styles[size],
    styles[color],
    {
      [styles.outline]: outline,
      [styles.disabled]: disabled,
    },
    className
  );

  return (
    <div className={outerClassName}>
      <ConditionalWrapper
        condition={linkProps !== undefined}
        wrapper={(children) => (
          <NextLink
            href={linkProps?.href ? linkProps.href : {}}
            {...linkProps}
            prefetch={false}
          >
            {children}
          </NextLink>
        )}
      >
        <button
          className={classnames(styles.btn, customStyles)}
          onClick={onClick}
          disabled={disabled}
        >
          {text}
        </button>
      </ConditionalWrapper>
    </div>
  );
};

export default ButtonRadius;
