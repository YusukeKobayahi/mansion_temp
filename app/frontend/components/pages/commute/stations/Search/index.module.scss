@import "~/styles/variables";
@import "~/styles/utils";

.aboveLabel {
  text-align: center;
  margin: 15px auto;
  color: #373a3c;
}
