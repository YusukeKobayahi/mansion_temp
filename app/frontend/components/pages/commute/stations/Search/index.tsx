import Router from "next/router";
import { useEffect, useState } from "react";
import { omitBy } from "lodash";
import {
  setInitialStateNumber,
  buildLibraryPagePath,
  yenFormat,
} from "~/lib/utils";
import {
  CommuteStationsQuery,
  CommuteStationsQueryVariables,
} from "~/graphql/generated";
import ButtonRect from "~/components/commons/ButtonRect";
import ButtonRadius, {
  ButtonRadiusProps,
} from "~/components/commons/ButtonRadius";
import SelectBox from "~/components/commons/SelectBox";
import BasicBlock from "~/components/molecules/Search/BasicBlock";
import SelectBlock from "~/components/molecules/Search/SelectBlock";
import DuritionSelectBlock from "~/components/molecules/Search/DuritionSelectBlock";
import Container from "~/components/molecules/Search/Container";

import styles from "./index.module.scss";

type SearchProps = {
  station: CommuteStationsQuery["station"];
  minimize?: boolean;
  searchCondition: CommuteStationsQueryVariables;
  buttonRadius: Pick<ButtonRadiusProps, "onClick" | "disabled">;
};

const Search: React.FC<SearchProps> = ({
  station,
  minimize = false,
  searchCondition,
  buttonRadius,
}: SearchProps) => {
  const [mnt, setMnt] = useState(setInitialStateNumber(searchCondition.mnt));
  const [trt, setTrt] = useState(setInitialStateNumber(searchCondition.trt));
  const [apb, setApb] = useState(setInitialStateNumber(searchCondition.apb));
  const [apt, setApt] = useState(setInitialStateNumber(searchCondition.apt));
  const [limit] = useState(setInitialStateNumber(searchCondition.limit));
  const [isOpen, setIsOpen] = useState(false);

  useEffect(() => {
    setIsOpen(false);
  }, [searchCondition]);

  const getNumArray = (start: number, end: number, step: number): number[] => {
    const array: number[] = [];
    for (let i = start; i <= end; i += step) {
      array.push(i);
    }
    return array;
  };
  const minutesTopOptions = getNumArray(5, 60, 5).map((val) => {
    return {
      value: String(val),
      text: String(val),
    };
  });
  minutesTopOptions.unshift({ value: "", text: "指定なし" });

  const trasferTimeTopOptions = getNumArray(1, 5, 1).map((val) => {
    return {
      value: String(val),
      text: String(val),
    };
  });
  trasferTimeTopOptions.unshift({ value: "", text: "指定なし" });

  const marketPriceOptions = getNumArray(1000000, 5000000, 1000000).map(
    (val) => {
      return {
        value: String(val),
        text: yenFormat(val),
      };
    }
  );
  const apbOptions = [{ text: "下限なし", value: "" }, ...marketPriceOptions];
  const aptOptions = [...marketPriceOptions, { text: "上限なし", value: "" }];

  const { orap, ortt, ormn } = searchCondition;
  const search = () => {
    const query = omitBy(
      {
        orap,
        ortt,
        ormn,
        mnt,
        trt,
        apb,
        apt,
        limit,
      },
      (value) => !value || value === ""
    );
    const href = {
      pathname: buildLibraryPagePath("commute", "stations", searchCondition.st),
      query,
    };
    Router.push(href);
  };

  return (
    <>
      <Container minimize={minimize} isOpen={isOpen} setIsOpen={setIsOpen}>
        <SelectBlock
          labelText={`${station?.name}駅までの時間`}
          unitText={"分圏内"}
        >
          <SelectBox
            options={minutesTopOptions}
            defaultValue={mnt}
            onBlur={(e) => setMnt(e.target.value)}
          />
        </SelectBlock>
        <SelectBlock labelText={"乗り換え回数"} unitText={"回以内"}>
          <SelectBox
            options={trasferTimeTopOptions}
            defaultValue={trt}
            onBlur={(e) => setTrt(e.target.value)}
          />
        </SelectBlock>
        <DuritionSelectBlock
          labelText={"平均相場(㎡)"}
          minSelectBox={
            <SelectBox
              options={apbOptions}
              defaultValue={apb}
              onBlur={(e) => setApb(e.target.value)}
            />
          }
          maxSelectBox={
            <SelectBox
              options={aptOptions}
              defaultValue={apt}
              onBlur={(e) => setApt(e.target.value)}
            />
          }
        />
        {/* sectionBarとBlockの非表示をレスポンシブ時に変更するための記述 */}
        <div className="spOnly">
          <BasicBlock sectionBar={false}>
            <ButtonRect text={"検索する"} outline={true} onClick={search} />
          </BasicBlock>
        </div>
        <div className="pcOnly">
          <BasicBlock>
            <ButtonRect text={"検索する"} outline={true} onClick={search} />
          </BasicBlock>
          <BasicBlock sectionBar={false}>
            <div className={styles.aboveLabel}>
              <p>駅を選択してください</p>
            </div>
            <ButtonRadius
              outline={true}
              text={"選択した駅の物件を見る"}
              disabled={buttonRadius.disabled}
              onClick={buttonRadius.onClick}
            />
          </BasicBlock>
        </div>
      </Container>
    </>
  );
};

export default Search;
