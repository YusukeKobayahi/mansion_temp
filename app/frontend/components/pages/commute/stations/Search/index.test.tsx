import { render } from "@testing-library/react";
import mockData from "../fixtures/mockData.json";
import { CommuteStationsQuery } from "~/graphql/generated";
import { ButtonRadiusProps } from "~/components/commons/ButtonRadius";
import Search from ".";

describe("components/pages/commute/stations/Search", () => {
  const mock = (mockData as unknown) as CommuteStationsQuery;
  const station = mock.station;
  const searchCondition = { st: "1322" };
  const buttonRadius: Pick<ButtonRadiusProps, "onClick" | "disabled"> = {
    disabled: false,
    onClick: () => {
      return;
    },
  };
  it("正常に表示される", () => {
    const { asFragment } = render(
      <Search
        station={station}
        searchCondition={searchCondition}
        buttonRadius={buttonRadius}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
