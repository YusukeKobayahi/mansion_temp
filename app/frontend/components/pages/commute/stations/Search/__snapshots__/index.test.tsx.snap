// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/pages/commute/stations/Search 正常に表示される 1`] = `
<DocumentFragment>
  <section>
    <div
      class="container"
    >
      <div
        class="pcOnly"
      >
        <div
          class="title"
        >
          <p>
            絞り込み条件
          </p>
        </div>
      </div>
      <div
        class="spOnly"
      >
        <button
          class="title"
        >
          <div
            style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
          >
            <div
              style="box-sizing: border-box; display: block; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjAiIGhlaWdodD0iMjAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
              />
            </div>
            <noscript />
            <img
              alt="option"
              data-nimg="true"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </div>
          <p>
            絞り込み条件
          </p>
          <div
            class="toggler"
          >
            <span
              class="close"
            />
          </div>
        </button>
      </div>
      <div
        class="body"
      >
        <div
          class="block sectionBar"
        >
          <div
            class="content"
          >
            <p
              class="label"
            >
              品川駅までの時間
            </p>
            <div>
              <div
                class="selectBox"
              >
                <select>
                  <option
                    selected=""
                    value=""
                  >
                    指定なし
                  </option>
                  <option
                    value="5"
                  >
                    5
                  </option>
                  <option
                    value="10"
                  >
                    10
                  </option>
                  <option
                    value="15"
                  >
                    15
                  </option>
                  <option
                    value="20"
                  >
                    20
                  </option>
                  <option
                    value="25"
                  >
                    25
                  </option>
                  <option
                    value="30"
                  >
                    30
                  </option>
                  <option
                    value="35"
                  >
                    35
                  </option>
                  <option
                    value="40"
                  >
                    40
                  </option>
                  <option
                    value="45"
                  >
                    45
                  </option>
                  <option
                    value="50"
                  >
                    50
                  </option>
                  <option
                    value="55"
                  >
                    55
                  </option>
                  <option
                    value="60"
                  >
                    60
                  </option>
                </select>
              </div>
            </div>
            <p
              class="unit"
            >
              分圏内
            </p>
          </div>
        </div>
        <div
          class="block sectionBar"
        >
          <div
            class="content"
          >
            <p
              class="label"
            >
              乗り換え回数
            </p>
            <div>
              <div
                class="selectBox"
              >
                <select>
                  <option
                    selected=""
                    value=""
                  >
                    指定なし
                  </option>
                  <option
                    value="1"
                  >
                    1
                  </option>
                  <option
                    value="2"
                  >
                    2
                  </option>
                  <option
                    value="3"
                  >
                    3
                  </option>
                  <option
                    value="4"
                  >
                    4
                  </option>
                  <option
                    value="5"
                  >
                    5
                  </option>
                </select>
              </div>
            </div>
            <p
              class="unit"
            >
              回以内
            </p>
          </div>
        </div>
        <div
          class="block sectionBar"
        >
          <div
            class="label"
          >
            <p>
              平均相場(㎡)
            </p>
          </div>
          <div
            class="inputs"
          >
            <div
              class="selectBox"
            >
              <select>
                <option
                  selected=""
                  value=""
                >
                  下限なし
                </option>
                <option
                  value="1000000"
                >
                  100万円
                </option>
                <option
                  value="2000000"
                >
                  200万円
                </option>
                <option
                  value="3000000"
                >
                  300万円
                </option>
                <option
                  value="4000000"
                >
                  400万円
                </option>
                <option
                  value="5000000"
                >
                  500万円
                </option>
              </select>
            </div>
            <span>
               ~ 
            </span>
            <div
              class="selectBox"
            >
              <select>
                <option
                  value="1000000"
                >
                  100万円
                </option>
                <option
                  value="2000000"
                >
                  200万円
                </option>
                <option
                  value="3000000"
                >
                  300万円
                </option>
                <option
                  value="4000000"
                >
                  400万円
                </option>
                <option
                  value="5000000"
                >
                  500万円
                </option>
                <option
                  selected=""
                  value=""
                >
                  上限なし
                </option>
              </select>
            </div>
          </div>
        </div>
        <div
          class="spOnly"
        >
          <div
            class="block"
          >
            <div
              class=""
            >
              <button
                class="btn brown outline"
              >
                検索する
              </button>
            </div>
          </div>
        </div>
        <div
          class="pcOnly"
        >
          <div
            class="block sectionBar"
          >
            <div
              class=""
            >
              <button
                class="btn brown outline"
              >
                検索する
              </button>
            </div>
          </div>
          <div
            class="block"
          >
            <div
              class="aboveLabel"
            >
              <p>
                駅を選択してください
              </p>
            </div>
            <div
              class=""
            >
              <button
                class="btn md orange outline"
              >
                選択した駅の物件を見る
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
